### PR TITLE
Allow round-tripping of the v2 lock file format in java

### PR DESCRIPTION
**Pull request by shs96c**
There have been several occasions where I need to get the information from the lock file in a reusable way. Having this available from within `rules_jvm_external`'s own classes would be most useful

### CODE REVIEW

File: tests/com/github/bazelbuild/rules_jvm_external/resolver/lockfile/V2LockFileTest.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+  @Test
+  public void shouldRoundTripM2Local() {
+    V2LockFile lockFile = new V2LockFile(repos, Set.of(), Set.of());
+    Map<String, Object> rendered = lockFile.render();
+    rendered.put("m2local", true);
+
+    V2LockFile roundTripped =
+        V2LockFile.create(
+            new GsonBuilder().setPrettyPrinting().serializeNulls().create().toJson(rendered));
```
Comment (jin) [inline tests/com/github/bazelbuild/rules_jvm_external/resolver/lockfile/V2LockFileTest.java:78]: Use the `roundTrip` helper?

File: private/tools/java/com/github/bazelbuild/rules_jvm_external/resolver/lockfile/V2LockFile.java [hunk 2]
```diff
@@ -12,32 +12,175 @@
...
+    return infos;
+  }
+
+  public Set<Conflict> getConflicts() {
+    return conflicts;
+  }
+
+  @SuppressWarnings("unchecked")
+  public static V2LockFile create(String from) {
+    URI m2LocalUri = Paths.get(USER_HOME.value()).resolve(".m2/repository").toUri();
```
Comment (jin) [inline private/tools/java/com/github/bazelbuild/rules_jvm_external/resolver/lockfile/V2LockFile.java:66]: nit: make this static and share it with `render`

File: private/tools/java/com/github/bazelbuild/rules_jvm_external/resolver/DependencyInfo.java [hunk 3]
```diff
@@ -81,6 +82,18 @@ public SortedMap<String, SortedSet<String>> getServices() {
     return services;
   }
 
+  @Override
+  public String toString() {
```
Comment (jin) [inline private/tools/java/com/github/bazelbuild/rules_jvm_external/resolver/DependencyInfo.java:86]: nit: use `MoreObjects#toStringHelper` which handles the formatting for you.

**Review by jin — APPROVED** [review]

File: private/tools/java/com/github/bazelbuild/rules_jvm_external/resolver/DependencyInfo.java (hunk 3 above)
Comment (shs96c) [reply private/tools/java/com/github/bazelbuild/rules_jvm_external/resolver/DependencyInfo.java:86]: Done.

File: private/tools/java/com/github/bazelbuild/rules_jvm_external/resolver/lockfile/V2LockFile.java (hunk 2 above)
Comment (shs96c) [reply private/tools/java/com/github/bazelbuild/rules_jvm_external/resolver/lockfile/V2LockFile.java]: Done

File: tests/com/github/bazelbuild/rules_jvm_external/resolver/lockfile/V2LockFileTest.java (hunk 1 above)
Comment (shs96c) [reply tests/com/github/bazelbuild/rules_jvm_external/resolver/lockfile/V2LockFileTest.java:76]: We amended the generated JSON by hand by adding `m2local` to it, so we can't use the `roundTrip` helper here. We _could_ have added the `M2_LOCAL_URI` to the list of repos when we create the lockfile instance, but I don't wanted to be sure we were honouring the `m2local` value in the rendered map properly.

**Pull request merged**